iree/hal/vmvx/registration/driver_module_sync.c

```diff
@@ -0,0 +1,81 @@
+// Copyright 2021 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree/hal/vmvx/registration/driver_module_sync.h"
+
+#include <inttypes.h>
+#include <stddef.h>
+
+#include "iree/base/api.h"
+#include "iree/hal/local/executable_loader.h"
+#include "iree/hal/local/loaders/vmvx_module_loader.h"
+#include "iree/hal/local/sync_device.h"
+#include "iree/hal/local/sync_driver.h"
+#include "iree/vm/api.h"
+
+// TODO(#4298): remove this driver registration and wrapper.
+
+// TODO(benvanik): replace with C flags.
+#define IREE_HAL_VMVX_WORKER_COUNT 0
+#define IREE_HAL_MAX_VMVX_WORKER_COUNT 16
+
+#define IREE_HAL_VMVX_SYNC_DRIVER_ID 0x53564D5658lu  // SVMVX
+
+static iree_status_t iree_hal_vmvx_sync_driver_factory_enumerate(
+    void* self, const iree_hal_driver_info_t** out_driver_infos,
+    iree_host_size_t* out_driver_info_count) {
+  static const iree_hal_driver_info_t driver_infos[1] = {
+      {
+          .driver_id = IREE_HAL_VMVX_SYNC_DRIVER_ID,
+          .driver_name = iree_string_view_literal("vmvx-sync"),
+          .full_name = iree_string_view_literal("VM-based reference backend"),
```

Comment:
could differentiate this full name from the name for vmvx
```suggestion
          .full_name = iree_string_view_literal("synchronous VM-based reference backend"),
```

Reply:
Done.